code/modules/projectiles/guns/projectile/shotgun.dm

```diff
@@ -156,6 +156,28 @@
 	ammo_type = /obj/item/ammo_casing/a762/enchanted
 
 
+/obj/item/weapon/gun/projectile/shotgun/boltaction/beebee
+	name = "\improper Red Rider BB gun"
+	desc = "You'll shoot your eye out kid!"
+	mag_type = /obj/item/ammo_box/magazine/internal/boltaction/beebee
+
+/obj/item/ammo_box/magazine/internal/boltaction/beebee
+	max_ammo = 50
+	ammo_type = /obj/item/ammo_casing/a762/beebee
+
+/obj/item/ammo_casing/a762/beebee
+	desc = "Bee Capsule. Oh god why"
+	icon_state = "762-casing"
+	caliber = "Angrybee"
+	projectile_type = /obj/item/projectile/bullet/beebee
+	
+/obj/item/projectile/bullet/beebee
+	damage = 5
+
+/obj/item/projectile/bullet/beebee/on_hit(atom/target, blocked = 0)
+	..()
+	new /mob/living/simple_animal/hostile/poison/bees(target.loc)

```

Comment:
Okay, get_turf() recursively gets the loc until it gets a turf.  Also, setting target is as easy as first istyping() the target(to make sure its a mob), then typecast a var to bees, bees.target = target.